### PR TITLE
Adding Fake Rate measurement machinery to HZA analysis

### PR DESCRIPTION
**Pull request by mmusich**
Title says it all.

### CODE REVIEW

File: bin/zhtautau/runZHTauTauAnalysis.cc [hunk 1]
```diff
@@ -1058,7 +1058,7 @@ int main(int argc, char* argv[])
     LorentzVector elDiff(0,0,0,0);
     for(size_t ilep=0; ilep<leptons.size(); ilep++){
       bool passKin(true),passId(true),passIso(true);
-      bool passLooseLepton(true), passSoftMuon(true), passSoftElectron(true), passVetoElectron(true);
+      bool passVeryLooseLepton(true), passLooseLepton(true), passSoftMuon(true), passSoftElectron(true), passVetoElectron(true);
```
Comment (mmusich) [inline bin/zhtautau/runZHTauTauAnalysis.cc:1061]: @clacaputo can you cross-check what I have done here correponds to what we discussed this morning? Thanks

File: bin/zhtautau/runZHTauTauAnalysis.cc [hunk 2]
```diff
@@ -1096,34 +1096,37 @@ int main(int argc, char* argv[])
...
+	    if(is2016MC){
+	      muCor2016->momcor_mc  (p4, lid<0 ? -1 :1, ntrk, qter);
+	    } else if (is2016data){   
+	      muCor2016->momcor_data(p4, lid<0 ? -1 :1, 0, qter);
+	    }
+	    
+	    
+	    muDiff -= leptons[ilep].p4();
+	    leptons[ilep].setP4(LorentzVector(p4.Px(),p4.Py(),p4.Pz(),p4.E() ) );
+	    muDiff += leptons[ilep].p4();
```
Comment (mmusich) [inline bin/zhtautau/runZHTauTauAnalysis.cc:1122]: shall we change the logic here as for the electrons?

```
leptons[ilep] = patUtils::GenericLepton(leptons[ilep].mu); //recreate the generic lepton to be sure that the p4 is ok
```

File: bin/zhtautau/runZHTauTauAnalysis.cc (hunk 1 above)
Comment (clacaputo) [reply bin/zhtautau/runZHTauTauAnalysis.cc:1061]: Yes, I will. Thanks

File: bin/zhtautau/runZHTauTauAnalysis.cc (hunk 2 above)
Comment (clacaputo) [reply bin/zhtautau/runZHTauTauAnalysis.cc:1122]: I have already added that line but in my branch. 
https://github.com/clacaputo/2l2v_fwk/blob/80X_Fixing/bin/zhtautau/runZHTauTauAnalysis.cc#L1010

I forgot to push it in the central repository

**Comment by mmusich** [comment]
Closing as it all entered in the repo via: https://github.com/cp3-lltautau/2l2v_fwk/pull/30

**Pull request closed without merging**